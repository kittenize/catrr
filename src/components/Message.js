import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const Container = glamorous.div({
    display: 'flex',
    flexDirection: 'column',
    backgroundColor: '#fff',
    boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.5)',
    transition: '0.3s',
    textAlign: 'left',
    padding: '2px 16px',
    margin: '15px',
    borderRadius: '5px',
});

const Row = glamorous.div({
    width: '100%',
    display: 'flex',
    alignItems: 'center'
});

const BoldSpan = glamorous.span({
    fontWeight: 'bold',
    paddingRight: '5px'
});

const SecondarySpan = glamorous.span({
    fontSize: '.7em',
    color: '#AAA'
});

const Message = ({username, date, message, id}) => (
    <Container>
        <Row>
            <BoldSpan>{username}</BoldSpan>
            <SecondarySpan>{(new Date(date)).toLocaleDateString()} {(new Date(date)).toLocaleTimeString()}</SecondarySpan>
        </Row>
        <Row>
            {message}
        </Row>
        <Row>
            <SecondarySpan>{id}</SecondarySpan>
        </Row>
    </Container>
);

Message.propTypes = {
    username: PropTypes.string,
    date: PropTypes.number,
    message: PropTypes.string,
    id: PropTypes.string
};


export default Message;