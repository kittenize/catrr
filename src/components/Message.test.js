import React from 'react';
import renderer from 'react-test-renderer';

import Message from './Message';

it('renders', () => {
    const tree = renderer
        .create(<Message username="cat" message="cat" date={new Date(1525931262185)} id="dog" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});