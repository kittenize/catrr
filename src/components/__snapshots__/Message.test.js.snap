// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="css-wikk8p"
>
  <div
    className="css-7zsmr0"
  >
    <span
      className="css-qdfjtt"
    >
      cat
    </span>
    <span
      className="css-1m45918"
    >
      2018-5-10
       
      01:47:42
    </span>
  </div>
  <div
    className="css-7zsmr0"
  >
    cat
  </div>
  <div
    className="css-7zsmr0"
  >
    <span
      className="css-1m45918"
    >
      dog
    </span>
  </div>
</div>
`;
